src/routes.tsx

```diff
@@ -41,13 +50,79 @@ import { ProposalDetailsPage } from './app/pages/ProposalDetailsPage'
 import { ConsensusBlocksPage } from './app/pages/ConsensusBlocksPage'
 import { ConsensusAccountsPage } from './app/pages/ConsensusAccountsPage'
 import { ConsensusTransactionsPage } from './app/pages/ConsensusTransactionsPage'
+import { FC, useEffect } from 'react'
 
 const NetworkSpecificPart = () => (
   <ThemeByNetwork network={useRequiredScopeParam().network}>
     <Outlet />
   </ThemeByNetwork>
 )
 
+/**
+ * Use the passed RouteObject unless we have a fixed layer.
+ *
+ * Normally, this will just wrap the RouteObject into an array.
+ * Unless the condition is true, because in that case, and empty array will be returned.
+ */
+const unlessFixedLayer = (route: RouteObject): RouteObject[] => (fixedLayer ? [] : [route])
+
+/**
+ * Use the passed RouteObject unless we have a fixed network and layer.
+ *
+ * Normally, this will just wrap the RouteObject into an array.
+ * Unless the condition is true, because in that case, and empty array will be returned.
+ */
+const unlessFixedNetworkAndLayer = (route: RouteObject): RouteObject[] =>
+  hasFixedNetworkAndLayer ? [] : [route]
+
+/**
+ * Use the passed RouteObject unless we are fixed on a ParaTime.
+ *
+ * Normally, this will just wrap the RouteObject into an array.
+ * Unless the condition is true, because in that case, and empty array will be returned.
+ */
+const unlessFixedOnParatime = (route: RouteObject): RouteObject[] => (isFixedOnParatime ? [] : [route])
+
+/**
+ * Use the passed RouteObject unless we are fixed on consensus.
+ *
+ * Normally, this will just wrap the RouteObject into an array.
+ * Unless the condition is true, because in that case, and empty array will be returned.
+ */
+const unlessFixedOnConsensus = (route: RouteObject): RouteObject[] => (isFixedOnConsensus ? [] : [route])
+
+/**
+ * Use the passed RouteObject if we have a fixed layer but no network
+ *
+ * Normally, this will just wrap the RouteObject into an array.
+ * Unless the condition is true, because in that case, and empty array will be returned.
+ */
+const ifFixedLayerButNoNetwork = (route: RouteObject): RouteObject[] =>
```

Comment:
I would skip `if` naming in a function.

Reply:
The semantics here is that we want to include this option only _if_ we have a fixed layer, but no (fixed) network.
So the function is not testing this condition, but doing something conditionally _if_ the condition is fulfilled.

What naming would you suggest?

Maybe `whenFixedLayerButNoNetwork()` ?

Reply:
Yeah, maybe the one you suggested, or `getFixedLayerButNoNetwork`. I don't know, but `if` feels to me counterintuitive.

Reply:
> > Maybe `whenFixedLayerButNoNetwork()` ?

> Yeah, maybe the one you suggested, 

OK, renamed.